File: backend/src/models/environmental-data.model.js
const mongoose = require('mongoose');

const environmentalDataSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  date: {
    type: Date,
    default: Date.now
  },
  temperature: {
    type: Number,
    min: -50,
    max: 50
  },
  humidity: {
    type: Number,
    min: 0,
    max: 100
  },
  lightIntensity: {
    type: Number,
    min: 0,
    max: 100
  },
  soundExposure: {
    type: String,
    enum: ['Silent', 'Quiet', 'Moderate', 'Loud', 'Very Loud']
  },
  noiseLevel: {
    type: Number,
    min: 0,
    max: 120
  },
  airQuality: {
    type: String,
    enum: ['Excellent', 'Good', 'Moderate', 'Poor', 'Very Poor']
  },
  sleepEnvironment: {
    type: String,
    enum: ['Bedroom', 'Living Room', 'Hotel', 'Other']
  },
  sleepPosition: {
    type: String,
    enum: ['Back', 'Side', 'Stomach', 'Multiple']
  },
  notes: {
    type: String,
    maxlength: 500
  }
}, {
  timestamps: true
});

// Index for faster queries
environmentalDataSchema.index({ userId: 1, date: -1 });

const EnvironmentalData = mongoose.model('EnvironmentalData', environmentalDataSchema);

module.exports = EnvironmentalData;

File: lib/models/environmental_data_model.dart
class EnvironmentalData {
  final double? temperature;
  final double? humidity;
  final double? lightIntensity;
  final String? soundExposure;
  final double? noiseLevel;
  final String? airQuality;
  final String? sleepEnvironment;
  final String? sleepPosition;
  final String? notes;

  EnvironmentalData({
    this.temperature,
    this.humidity,
    this.lightIntensity,
    this.soundExposure,
    this.noiseLevel,
    this.airQuality,
    this.sleepEnvironment,
    this.sleepPosition,
    this.notes,
  });

  Map<String, dynamic> toJson() {
    return {
      'temperature': temperature,
      'humidity': humidity,
      'lightIntensity': lightIntensity,
      'soundExposure': soundExposure,
      'noiseLevel': noiseLevel,
      'airQuality': airQuality,
      'sleepEnvironment': sleepEnvironment,
      'sleepPosition': sleepPosition,
      'notes': notes,
    };
  }

  factory EnvironmentalData.fromJson(Map<String, dynamic> json) {
    return EnvironmentalData(
      temperature: json['temperature']?.toDouble(),
      humidity: json['humidity']?.toDouble(),
      lightIntensity: json['lightIntensity']?.toDouble(),
      soundExposure: json['soundExposure'],
      noiseLevel: json['noiseLevel']?.toDouble(),
      airQuality: json['airQuality'],
      sleepEnvironment: json['sleepEnvironment'],
      sleepPosition: json['sleepPosition'],
      notes: json['notes'],
    );
  }

  static List<String> get soundExposureOptions => [
    'Silent',
    'Quiet',
    'Moderate',
    'Loud',
    'Very Loud'
  ];

  static List<String> get airQualityOptions => [
    'Excellent',
    'Good',
    'Moderate',
    'Poor',
    'Very Poor'
  ];

  static List<String> get sleepEnvironmentOptions => [
    'Bedroom',
    'Living Room',
    'Hotel',
    'Other'
  ];

  static List<String> get sleepPositionOptions => [
    'Back',
    'Side',
    'Stomach',
    'Multiple'
  ];
}
